File: src/main/java/dev/macrohq/swiftslayer/util/Ref.java
package dev.macrohq.swiftslayer.util;

import dev.macrohq.swiftslayer.config.SwiftSlayerConfig;
import net.minecraft.client.Minecraft;
import net.minecraft.client.entity.EntityPlayerSP;
import net.minecraft.client.settings.GameSettings;
import net.minecraft.world.World;

public class Ref {
    public static Minecraft mc() { return Minecraft.getMinecraft(); }
    public static EntityPlayerSP player() { return mc().thePlayer; }
    public static World world() { return mc().theWorld; }
    public static GameSettings gameSettings() { return mc().gameSettings; }
}

File: src/main/kotlin/dev/macrohq/swiftslayer/util/Ref.kt
package dev.macrohq.swiftslayer.util

import dev.macrohq.swiftslayer.SwiftSlayer
import dev.macrohq.swiftslayer.macro.mobKillers.RevMobKiller
import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.Tessellator

val mc
    get() = Minecraft.getMinecraft()
val player
    get() = mc.thePlayer
val world
    get() = mc.theWorld
val tessellator
    get() = Tessellator.getInstance()
val swiftSlayer
    get() = SwiftSlayer
val gameSettings
    get() = mc.gameSettings
val config
    get() = swiftSlayer.config
val pathExecutor
    get() = swiftSlayer.pathExecutor
val mobKiller
    get() = when(SwiftSlayer.config.slayer) {
            0 -> {
                RevMobKiller
            }

        else -> {
            RevMobKiller
        }
    }
val autoBatphone
    get() = swiftSlayer.autoBatphone
val macroManager
    get() = swiftSlayer.macroManager
val genericBossKiller
    get() = RevMobKiller.getInstance()
val endermanBossKiller
    get() = swiftSlayer.endermanBossKiller
val revenant
    get() = swiftSlayer.revenant
val tracker
    get() = swiftSlayer.tracker